sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -280,6 +280,60 @@ object GpuOrcScan extends Arm {
   def testLongMultiplicationOverflow(a: Long, b: Long) = {
     Math.multiplyExact(a, b)
   }
+
+
+  /**
+   * Convert the integer vector into timestamp(microseconds) vector.
+   * @param col The integer columnar vector.
+   * @param colType Specific integer type, it should be BOOL/INT8/INT16/INT32/INT64.
+   * @param asSeconds True if consider the integers as seconds, otherwise we consider them
+   *                  as milliseconds. This parameter is determined by the shims.
+   * @return A timestamp vector.
+   */
+  def castIntegersToTimestamp(col: ColumnView, colType: DType,
+                              asSeconds: Boolean = true): ColumnVector = {
+    assert(colType == DType.BOOL8 || colType == DType.INT8 || colType == DType.INT16
+      || colType == DType.INT32 || colType == DType.INT64)
+
+    colType match {
+      case DType.BOOL8 | DType.INT8 | DType.INT16 | DType.INT32 =>
+        // Convert seconds to microseconds, we need to multiply 1e36
+        // Convert milliseconds to microseconds, we need to multiply 1e3
+        // integers * value => microseconds
+        val value = if (asSeconds) { 1000000 } else { 1000 }
```

Comment:
The name value is too generic. Maybe `microsecodndsFactor`? 

You can also avoid hardcoding if you switch to TimeUnit the tune of `TimeUnit.MICROSECONDS.convert(1L, TimeUnit.SECONDS)`

Reply:
We can use `DateTimeConstants.MICROS_PER_MILLIS`